### PR TITLE
Add support for deterministic build for msbuild driver

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
contributes to https://github.com/tonerdo/coverlet/issues/363

@clairernovotny @tmat @tonerdo the main files to check are:

DeterministicBuild.targets -> workaround
src/coverlet.core/Helpers/SourceRootTranslator.cs -> path translator
test/coverlet.core.tests/Helpers/SourceRootTranslatorTests.cs -> tests
src/coverlet.msbuild.tasks/coverlet.msbuild.targets -> target for tests project 

All other updates are tests fix and DI container update.

### CODE REVIEW

**Comment by clairernovotny** [comment]
Where is `DeterministicBuild.targets` being used? Just for the local project?

There'll need to be a doc update that tells people they'll need that in their `Directory.Build.targets` until some future SDK update.

**Comment by MarcoRossignoli** [comment]
> Where is DeterministicBuild.targets being used? Just for the local project?

Is imported here https://github.com/tonerdo/coverlet/pull/796/files#diff-56f8482aeb786dd105fbf93dedf2615f

> There'll need to be a doc update that tells people they'll need that in their Directory.Build.targets until some future SDK update.

Yes it's the last point in the list https://github.com/tonerdo/coverlet/issues/363#issuecomment-609055100 I'll add samples for msbuild and collector here https://github.com/tonerdo/coverlet/blob/master/Documentation/Examples.md before release package.

File: src/coverlet.msbuild.tasks/coverlet.msbuild.targets [hunk 1]
```diff
@@ -3,6 +3,21 @@
   <UsingTask TaskName="Coverlet.MSbuild.Tasks.InstrumentationTask" AssemblyFile="$(CoverletToolsPath)coverlet.msbuild.tasks.dll"/>
   <UsingTask TaskName="Coverlet.MSbuild.Tasks.CoverageResultTask" AssemblyFile="$(CoverletToolsPath)coverlet.msbuild.tasks.dll"/>
 
+  <Target Name="ReferencedPathMaps" BeforeTargets="CoreCompile" >
```
Comment (clairernovotny) [inline src/coverlet.msbuild.tasks/coverlet.msbuild.targets:6]: Task needs the `Outputs` attribute with the file path that you're writing. That'll ensure that it's added to the list of files for clean.

See example: https://github.com/dotnet/sdk/blob/e605483699158a9202beb4eb9b4e18717c83a6f9/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.GenerateAssemblyInfo.targets#L141

Comment (MarcoRossignoli) [reply src/coverlet.msbuild.tasks/coverlet.msbuild.targets:6]: ok, thanks.

Comment (MarcoRossignoli) [reply src/coverlet.msbuild.tasks/coverlet.msbuild.targets:6]: @clairernovotny what do you think about naming target `CoverletReferencedPathMaps`?

Comment (MarcoRossignoli) [reply src/coverlet.msbuild.tasks/coverlet.msbuild.targets:6]: Actually also other targets doesn't prefix with `Coverlet` so never mind.

Comment (MarcoRossignoli) [reply src/coverlet.msbuild.tasks/coverlet.msbuild.targets:6]: I tried with output but it doesn't work so after some research I found an article https://docs.microsoft.com/en-us/archive/msdn-magazine/2009/february/msbuild-best-practices-for-creating-reliable-builds-part-1#id0090093 that describe `FileWrites` item list and seems to works.

**Comment by MarcoRossignoli** [comment]
@clairernovotny @tmat let me know if the translation alg and props/targets are ok to you and I'll take another review round and after I'll merge and I'll go ahead with the todo list.

**Comment by MarcoRossignoli** [comment]
@clairernovotny tomorrow if you want you can try with nightly build, it's the most "official" after nuget release https://github.com/tonerdo/coverlet/blob/master/Documentation/ConsumeNightlyBuild.md

**Pull request merged**